quickstarts/microsoft.storagepool/diskpool-create-entry-level/azuredeploy.json

```diff
@@ -0,0 +1,163 @@
+        "diskPoolTier": {
+            "type": "string",
+            "defaultValue": "Standard",
+            "allowedValues": [
+                "Standard"
+            ],
+            "metadata": {
+                "description": "Tier of the Disk Pool"
+            }
+        },
+        "diskPoolLocation": {
+            "type": "string",
+            "allowedValues": [
+                "AustraliaEast",
+                "CanadaCentral",
+                "CentralUS",
+                "EastUS",
+                "JapanEast",
+                "NorthEurope",
+                "SouthCentralUS",
+                "SoutheastAsia",
+                "UKSouth",
+                "WestEurope",
+                "WestUS2"
+            ],
+            "metadata": {
+                "description": "Location of the Disk Pool"
+            }
+        },
+        "managedDiskName": {
+            "defaultValue": "disk-01",
+            "type": "string",
+            "metadata": {
+                "description": "Name of the managed disk (512 sector size) to create and export as an iSCSI LUN in the Disk Pool"
+            }
+        },
+        "availabilityZone": {
+            "defaultValue": 1,
+            "type": "string",
+            "allowedValues": [
+                "1",
+                "2",
+                "3"
+            ],
+            "metadata": {
+                "description": "Availability zone to deploy the managed disk and Disk Pool"
+            }
+        },
+        "diskPoolName": {
+            "defaultValue": "diskpool-01",
+            "type": "string",
+            "metadata": {
+                "description": "Name of the Disk Pool"
+            }
+        },
+        "targetName": {
+            "defaultValue": "iscsi-target-01",
+            "type": "string",
+            "metadata": {
+                "description": "Name of the iSCSI Target"
+            }
+        },
+        "existingVnetResourceGroupName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the resourceGroup for the existing virtual network to deploy the Disk Pool into."
+            }
+        },
+        "existingVnetName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the existing virtual network to deploy the Disk Pool into."
+            }
+        },
+        "existingSubnetName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the existing subnet to deploy the Disk Pool into."
+            }
+        }
+    },
+    "variables": {
+        "diskId": "[resourceId('Microsoft.Compute/disks/', parameters('managedDiskName'))]",
+        "diskRoleDefinitionName" : "[guid(resourceGroup().id, deployment().name)]",
+        "virtualMachineContributorRbacRoleDefinitionId": "9980e02c-c2be-4d73-94e8-173b1dc7cf3c",
+        "storagePoolRoleDefinitionId": "5741a1ff-751d-4ad7-bcd1-dfe3c998fd11"
```

Comment:
I believe this must be set to the id field from this command:
 Get-AzADServicePrincipal -DisplayName "StoragePool Resource Provider"

Reply:
Discussed offline. The Disk Pool operator role needs to be set to the StoragePool Resource Provider to attach the disk to the diskpool.